source/change-streams/tests/change-streams-resume-errorLabels.yml

```diff
@@ -0,0 +1,1134 @@
+# Tests for resume behavior on server versions that support the ResumableChangeStreamError label
+collection_name: &collection_name "test"
+database_name: &database_name "change-stream-tests"
+collection2_name: &collection2_name "test2"
+database2_name: &database2_name "change-stream-tests-2"
+tests:
+  -
+    description: "change stream resumes after HostUnreachable"
+    minServerVersion: "4.3.1"
+    failPoint:
+      configureFailPoint: failGetMoreAfterCursorCheckout # SERVER-46091 explains why a new failpoint was needed
+      mode: { times: 1 }
+      data:
+          errorCode: 6
+          closeConnection: false
+    target: collection
+    topology: 
+      - replicaset
+      - sharded
+    changeStreamPipeline: []
+    changeStreamOptions: {}
+    operations:
+      -
+        database: *database_name
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            x: 1
+    expectations:
+      -
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            cursor: {}
+            pipeline:
+              -
+                $changeStream: {}
+          command_name: aggregate
+          database_name: *database_name
+      -
+        command_started_event:
+          command:
+            getMore: "42"
+            collection: *collection_name
+          command_name: getMore
+          database_name: *database_name
+      -
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            cursor: {}
+            pipeline:
+              -
+                $changeStream: {}
+    result:
+      success:
+        -
+          _id: "42"
+          documentKey: "42"
+          operationType: insert
+          ns:
+            db: *database_name
+            coll: *collection_name
+          fullDocument:
+            x:
+              $numberInt: "1"
+  -
+    description: "change stream resumes after HostNotFound"
```

Comment:
If the resuming tests on whitelisted errors are all the same except for the failpoint error code and description, you could use YAML aliases to deduplicate. Example: https://gist.github.com/kevinAlbs/5b520389a0509c2544fbb8633ba18cf3#file-anchors-yml-L66-L70
But I'm ok with this as-is too.

Reply:
Maybe I just don't have enough experience reading YAML, but I find the tests more readable this way. Open to changing if others agree though.